Add support for security logging plugin

kyrofa opened the pull request:
If a `logging.xml` file is included in the security files, this PR enables the security logging plugin in Fast RTPS using the properties contained in said file.

kyrofa commented on rmw_fastrtps_shared_cpp/src/rmw_security_logging.cpp:
```diff
@@ -0,0 +1,196 @@
+const char distribute_enable_property_name[] =
+  "dds.sec.log.builtin.DDS_LogTopic.distribute";
+const char distribute_depth_property_name[] =
+  "com.rti.serv.secure.logging.distribute.writer_history_depth";
```
Obviously this is wrong, just waiting on more concrete QoS details.

kyrofa commented on rmw_fastrtps_shared_cpp/src/rmw_security_logging.cpp:
```diff
@@ -0,0 +1,196 @@
+void add_properties_from_qos_profile(
+  eprosima::fastrtps::rtps::PropertySeq & properties, const rmw_qos_profile_t & profile)
+{
+  // TODO(kyrofa): Talk to artivis about what QoS settings will be exposed.
```
@artivis have you done enough investigation into the low-level API to know what QoS knobs we can turn, yet? If so we can probably hammer out the design right here. Otherwise this can wait.

artivis replied:
Not yet, I'll get onto this first thing next week.

kyrofa replied:
We can wait for a follow-up PR. This one was getting big anyway.

artivis commented on rmw_fastrtps_shared_cpp/src/rmw_security.cpp:
```diff
@@ -0,0 +1,368 @@
+    properties,
+    verbosity_property_name,
+    *log_element,
+    "verbosity");
```
At the moment this does not append the `_LEVEL` suffix we discussed.

kyrofa replied:
Ah indeed, thanks for the reminder.

kyrofa replied:
Done.

ivanpauno committed:
Correct error message when event is not supported (#358)

Signed-off-by: Ivan Santiago Paunovic <ivanpauno@ekumenlabs.com>

kyrofa committed:
Add support for security logging plugin

If a `logging.xml` file is included in the security files, enable the
security logging plugin in Fast RTPS using the properties contained in
said file.

Also reorganize existing security logic to be more testable, and add
some tests for it.

Signed-off-by: Kyle Fazzari <kyle@canonical.com>
